src/js/vendor/docsearch.bundle.js

```diff
@@ -1,39 +1,63 @@
-    apiKey: config.apiKey,
-    indexName: config.indexName,
-    inputSelector: '#search-input',
-    autocompleteOptions: { hint: false, keyboardShortcuts: ['s'] },
-    algoliaOptions: defaultAlgoliaOptions,
-  })
-  var search = docsearchComponent.autocomplete
-  search.on('autocomplete:updated', function () {
-    this.scrollTop = 0
-  }.bind(search.getWrapper().firstChild))
-  search.on('autocomplete:closed', function () {
-    this.setVal()
-  }.bind(search))
+  activateSearch(require('docsearch.js/dist/cdn/docsearch.js'), document.getElementById('search-script').dataset)
 
-  var searchGlobalCheckbox = document.getElementById('search-global')
-  if (searchGlobalCheckbox && searchGlobalCheckbox.dataset && searchGlobalCheckbox.dataset.component) {
-    searchGlobalCheckbox.addEventListener('change', function () {
-      if (searchGlobalCheckbox.checked) {
-        docsearchComponent.algoliaOptions = defaultAlgoliaOptions
-      } else {
-        docsearchComponent.algoliaOptions = Object.assign(defaultAlgoliaOptions, {
-          facetFilters: ['component:' + searchGlobalCheckbox.dataset.component],
-        })
-      }
+  function activateSearch (docsearch, config) {
+    appendStylesheet(config.stylesheet)
+    var algoliaOptions = { hitsPerPage: parseInt(config.maxResults) || 15 }
+    var searchFieldSelector = '#' + (config.searchFieldId || 'search')
+    var searchField = document.querySelector(searchFieldSelector)
+    var filterInput = searchField.querySelector('.filter input')
+    var controller = docsearch({
+      appId: config.appId,
+      apiKey: config.apiKey,
+      indexName: config.indexName,
+      inputSelector: searchFieldSelector + ' .query',
+      autocompleteOptions: { debug: true, hint: false, keyboardShortcuts: ['s'], minLength: 2 },
+      algoliaOptions: algoliaOptions,
+      queryHook:
+        filterInput &&
+        function (query) {
+          controller.algoliaOptions = filterInput.checked
+            ? Object.assign({}, algoliaOptions, { facetFilters: [filterInput.dataset.facetFilter] })
+            : algoliaOptions
+        },
     })
+    var eventEmitter = controller.autocomplete
+    var autocomplete = eventEmitter.autocomplete
+    eventEmitter.on('autocomplete:updated', resetScroll.bind(autocomplete.getWrapper().firstChild))
+    if (filterInput) filterInput.addEventListener('change', toggleFilter.bind(controller.input))
+    searchField.addEventListener('click', confineEvent)
+    document.documentElement.addEventListener('click', resetSearch.bind(autocomplete))
+    if (controller.input.attr('autofocus')) controller.input.focus()
+  }
+
+  function appendStylesheet (href) {
+    var link = document.createElement('link')
+    link.rel = 'stylesheet'
+    link.href = href
+    document.head.appendChild(link)
+  }
+
+  function resetScroll () {
+    this.scrollTop = 0
+  }
+
+  function toggleFilter () {
+    this.focus()
+    var dropdown = this.data('aaAutocomplete').dropdown
```

Comment:
I can't see this data attribute :thinking:

Reply:
It's not a data attribute. It's a zepto object attribute. It's where the autocomplete library gets attached in the DOM. We have to go low level to get around the query cache in DocSearch.

Reply:
I spent somewhere between 8 - 12 hours to figure out why DocSearch was not running a new query when the filter was toggled. It's because it does not consider the facet filter in the query cache.